--- Rendering\ANARI\vtkAnariPolyDataMapperInheritInterface.cxx	2026-02-26 12:51:15.000000000 -0800
+++ Rendering\ANARI\vtkAnariPolyDataMapperInheritInterface.cxx	2026-02-26 13:23:12.000000000 -0800
@@ -30,37 +30,37 @@
   return ParameterFlags();
 }
 
 //----------------------------------------------------------------------------
 anari::Geometry vtkAnariPolyDataMapperInheritInterface::InitializeSpheres(vtkPolyData*,
   vtkProperty*, std::vector<vec3>&, std::vector<uint32_t>&, double, vtkDataArray*,
-  vtkPiecewiseFunction*, std::vector<vec2>&, std::vector<float>&, std::vector<vec4>&, int)
+  vtkPiecewiseFunction*, std::vector<vec2>&, std::vector<vec2>&, std::vector<vec4>&, int)
 {
   return anari::newObject<anari::Geometry>(this->AnariDevice, "sphere");
 }
 
 //----------------------------------------------------------------------------
 anari::Geometry vtkAnariPolyDataMapperInheritInterface::InitializeCurves(vtkPolyData*, vtkProperty*,
   std::vector<vec3>&, std::vector<uint32_t>&, double, vtkDataArray*, vtkPiecewiseFunction*,
-  std::vector<vec2>&, std::vector<float>&, std::vector<vec4>&, int)
+  std::vector<vec2>&, std::vector<vec2>&, std::vector<vec4>&, int)
 {
   return anari::newObject<anari::Geometry>(this->AnariDevice, "curve");
 }
 
 //----------------------------------------------------------------------------
 anari::Geometry vtkAnariPolyDataMapperInheritInterface::InitializeCylinders(vtkPolyData*,
   vtkProperty*, std::vector<vec3>&, std::vector<uint32_t>&, double, vtkDataArray*,
-  vtkPiecewiseFunction*, std::vector<vec2>&, std::vector<float>&, std::vector<vec4>&, int)
+  vtkPiecewiseFunction*, std::vector<vec2>&, std::vector<vec2>&, std::vector<vec4>&, int)
 {
   return anari::newObject<anari::Geometry>(this->AnariDevice, "cylinder");
 }
 
 //----------------------------------------------------------------------------
 anari::Geometry vtkAnariPolyDataMapperInheritInterface::InitializeTriangles(vtkPolyData*,
   vtkProperty*, std::vector<vec3>&, std::vector<uint32_t>&, std::vector<vec3>&, std::vector<vec2>&,
-  std::vector<float>&, std::vector<vec4>&, int)
+  std::vector<vec2>&, std::vector<vec4>&, int)
 {
   return anari::newObject<anari::Geometry>(this->AnariDevice, "triangle");
 }
 
 //----------------------------------------------------------------------------
 const char* vtkAnariPolyDataMapperInheritInterface::GetSpheresPostfix() const
