--- Rendering\ANARI\vtkAnariVolumeMapper.h	2025-06-23 11:12:36.000000000 -0800
+++ Rendering\ANARI\vtkAnariVolumeMapper.h	2025-09-17 14:50:54.000000000 -0800
@@ -49,12 +49,17 @@
   void Render(vtkRenderer*, vtkVolume*) override;
 
   /**
    * Allow vtkAnariSceneGraph properties to be set on the internal vtkRenderer.
    */
   vtkRenderer* GetInternalRenderer() const { return this->InternalRenderer; }
+  
+  /**
+   * Get the internal ANARI render pass.
+   */
+  vtkAnariPass* GetAnariPass() const { return this->InternalAnariPass; }
 
   //@{
   /**
    * Set/Get whether ANARI has been initialized.
    * By default, Initialized is false.
    */
