--- Rendering\ANARI\vtkAnariSceneGraph.cxx	2026-02-26 12:51:15.000000000 -0800
+++ Rendering\ANARI\vtkAnariSceneGraph.cxx	2026-02-26 13:24:37.000000000 -0800
@@ -224,22 +224,24 @@
 
 //----------------------------------------------------------------------------
 void vtkAnariSceneGraph::UpdateAnariFrameSize()
 {
   const uvec2 frameSize = { static_cast<uint32_t>(this->Size[0]),
     static_cast<uint32_t>(this->Size[1]) };
-  if ((uint32_t)this->Internal->ImageX == frameSize[0] &&
-    (uint32_t)this->Internal->ImageY == frameSize[1])
+  const size_t totalSize = this->Size[0] * this->Size[1];
+  if ((uint32_t)this->Internal->ImageY == frameSize[1] && !this->Internal->ColorBuffer.empty() &&
+    !this->Internal->DepthBuffer.empty() &&
+    this->Internal->ColorBuffer.size() == totalSize * sizeof(float) &&
+    this->Internal->DepthBuffer.size() == totalSize)
   {
     return;
   }
 
   this->Internal->ImageX = frameSize[0];
   this->Internal->ImageY = frameSize[1];
-
-  const size_t totalSize = this->Size[0] * this->Size[1];
+  
   this->Internal->ColorBuffer.resize(totalSize * sizeof(float));
   this->Internal->DepthBuffer.resize(totalSize);
 
   auto anariDevice = this->GetDeviceHandle();
   auto anariFrame = this->Internal->AnariFrame;
   anari::setParameter(anariDevice, anariFrame, "size", frameSize);
