--- tools\silex\CMakeLists.txt	2025-11-20 21:30:59.000000000 -0800
+++ tools\silex\CMakeLists.txt	2026-03-02 08:54:05.000000000 -0800
@@ -114,14 +114,14 @@
     # creates tmp dir used for installation
     # also deploys to build dir so silex can be run from build
     add_custom_command(TARGET silex
         POST_BUILD
         COMMAND ${CMAKE_COMMAND} -E remove_directory "${CMAKE_CURRENT_BINARY_DIR}/windeployqt"
         COMMAND set PATH=%PATH%$<SEMICOLON>${qt5_install_prefix}/bin
-        COMMAND Qt6::windeployqt --dir "${CMAKE_CURRENT_BINARY_DIR}/windeployqt" "$<TARGET_FILE:silex>"
-        COMMAND Qt6::windeployqt --verbose 0 --dir "${Silo_BINARY_DIR}/bin/$<CONFIG>" "$<TARGET_FILE:silex>"
+        COMMAND Qt6::windeployqt --verbose 0 --no-compiler-runtime --no-system-d3d-compiler --release --dir "${CMAKE_CURRENT_BINARY_DIR}/windeployqt" "$<TARGET_FILE:silex>"
+        COMMAND Qt6::windeployqt --verbose 0 --no-compiler-runtime --no-system-d3d-compiler --release --dir "${Silo_BINARY_DIR}/bin/$<CONFIG>" "$<TARGET_FILE:silex>"
     )
 
     # copy deployment directory during installation
     install(
         DIRECTORY
         "${CMAKE_CURRENT_BINARY_DIR}/windeployqt/"
