--- src\silo\silo.c
+++ src\silo\silo.c
@@ -11385,12 +11385,12 @@ include_point(int ptidx, int ndims, int const *dims,
     int j = ndims>1 ? (ptidx/dims[0])           % dims[1] : 0;
     int k = ndims>2 ? (ptidx/(dims[1]*dims[0])) % dims[2] : 0;

-    if (i < minidx[0]) return 0;
-    if (i > maxidx[0]) return 0;
-    if (j < minidx[1]) return 0;
-    if (j > maxidx[1]) return 0;
-    if (k < minidx[2]) return 0;
-    if (k > maxidx[2]) return 0;
+    if (ndims > 0 && i < minidx[0]) return 0;
+    if (ndims > 0 && i > maxidx[0]) return 0;
+    if (ndims > 1 && j < minidx[1]) return 0;
+    if (ndims > 1 && j > maxidx[1]) return 0;
+    if (ndims > 2 && k < minidx[2]) return 0;
+    if (ndims > 2 && k > maxidx[2]) return 0;

     return 1;
 }
