--- CMake\SiloConfig.cmake.in	2025-11-20 21:30:59.000000000 -0800
+++ CMake\SiloConfig.cmake.in	2026-03-02 08:00:00.000000000 -0800
@@ -96,15 +96,15 @@
 
 #-----------------------------------------------------------------------------
 # Don't include targets if this file is being picked up by another
 # project which has already built SILO as a subproject
 #-----------------------------------------------------------------------------
 if(NOT TARGET @SILO_NAME@)
-  include (${PACKAGE_PREFIX_DIR}/lib/cmake/Silo/@silo_targets_name@.cmake)
+  include (${CMAKE_CURRENT_LIST_DIR}/@silo_targets_name@.cmake)
 
   if(SILO_ENABLE_HDF5)
       include(CMakeFindDependencyMacro)
       find_dependency(hdf5 @HDF5_VERSION@ PATHS ${HDF5_DIR})
   endif()
 endif()
 
 
