--- CMakeInstallation.cmake	2025-11-10 16:00:49.000000000 -0800
+++ CMakeInstallation.cmake	2025-12-09 12:47:19.000000000 -0800
@@ -159,17 +159,17 @@
   set (BYTESEX little-endian)
 endif ()
 configure_file (
     ${HDF5_SOURCE_DIR}/src/libhdf5.settings.in
     ${HDF5_SRC_BINARY_DIR}/libhdf5.settings ESCAPE_QUOTES @ONLY
 )
-install (
-    FILES ${HDF5_SRC_BINARY_DIR}/libhdf5.settings
-    DESTINATION ${HDF5_INSTALL_LIB_DIR}
-    COMPONENT libraries
-)
+#install (
+#    FILES ${HDF5_SRC_BINARY_DIR}/libhdf5.settings
+#    DESTINATION ${HDF5_INSTALL_LIB_DIR}
+#    COMPONENT libraries
+#)
 
 #-----------------------------------------------------------------------------
 # Configure the HDF5_Examples.cmake file and the examples
 #-----------------------------------------------------------------------------
 option (HDF5_PACK_EXAMPLES  "Package the HDF5 Library Examples Compressed File" OFF)
 if (HDF5_PACK_EXAMPLES)
@@ -190,18 +190,18 @@
   install (
     DIRECTORY ${HDF5_SOURCE_DIR}/HDF5Examples
     DESTINATION ${HDF5_INSTALL_DATA_DIR}
     USE_SOURCE_PERMISSIONS
     COMPONENT hdfdocuments
   )
-  install (
-      FILES
-          ${HDF5_SOURCE_DIR}/release_docs/USING_CMake_Examples.txt
-      DESTINATION ${HDF5_INSTALL_DATA_DIR}
-      COMPONENT hdfdocuments
-  )
+  #install (
+  #    FILES
+  #        ${HDF5_SOURCE_DIR}/release_docs/USING_CMake_Examples.txt
+  #    DESTINATION ${HDF5_INSTALL_DATA_DIR}
+  #    COMPONENT hdfdocuments
+  #)
   install (
       FILES
           ${HDF_CONFIG_DIR}/examples/CTestScript.cmake
       DESTINATION ${HDF5_INSTALL_DATA_DIR}
       COMPONENT hdfdocuments
   )
@@ -268,17 +268,17 @@
         set (release_files
             ${release_files}
             ${HDF5_SOURCE_DIR}/release_docs/README_HPC.md
         )
       endif ()
     endif ()
-    install (
-        FILES ${release_files}
-        DESTINATION ${HDF5_INSTALL_DOC_DIR}
-        COMPONENT hdfdocuments
-    )
+    #install (
+    #    FILES ${release_files}
+    #    DESTINATION ${HDF5_INSTALL_DOC_DIR}
+    #    COMPONENT hdfdocuments
+    #)
   endif ()
 endif ()
 
 #-----------------------------------------------------------------------------
 # Set the cpack variables
 #-----------------------------------------------------------------------------
@@ -486,13 +486,13 @@
   endif ()
 
   # By default, do not warn when built on machines using only VS Express:
   if (NOT DEFINED CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_NO_WARNINGS)
     set (CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_NO_WARNINGS ON)
   endif ()
-  include (InstallRequiredSystemLibraries)
+  #include (InstallRequiredSystemLibraries)
 
   set (CPACK_INSTALL_CMAKE_PROJECTS "${HDF5_BINARY_DIR};HDF5;ALL;/")
 
   if (HDF5_PACKAGE_EXTLIBS)
     if (HDF5_ALLOW_EXTERNAL_SUPPORT MATCHES "GIT" OR HDF5_ALLOW_EXTERNAL_SUPPORT MATCHES "TGZ")
       if (H5_ZLIB_FOUND AND ZLIB_USE_EXTERNAL)
