--- libsrc\XdmfValuesBinary.cxx	2011-03-10 17:45:29.000000000 -0800
+++ libsrc\XdmfValuesBinary.cxx	2021-03-04 16:20:29.000000000 -0800
@@ -279,13 +279,13 @@
     if( ( strlen( this->DOM->GetWorkingDirectory() ) > 0 ) && 
       ( DataSetName[0] != '/' ) ){
         FullFileName << this->DOM->GetWorkingDirectory() << "/";
     }
     FullFileName << DataSetName << ends;
     char * path = FullFileName.rdbuf()->str();
-    XdmfDebug("Opening Binary Data for Reading : " << FullFileName);
+    XdmfDebug("Opening Binary Data for Reading : " << path);
 
 
     //char * path = new char [ strlen(this->DOM->GetWorkingDirectory())+strlen(DataSetName) + 1 ];
     //strcpy(path, this->DOM->GetWorkingDirectory());
     //strcpy(path+strlen(this->DOM->GetWorkingDirectory()), DataSetName);
     try{
