--- src\cgnstypes.h.in	2026-02-18 15:01:18.000000000 -0800
+++ src\cgnstypes.h.in	2026-02-19 12:15:59.000000000 -0800
@@ -36,17 +36,12 @@
 #define CG_BUILD_BASESCOPE @BUILDBASESCOPE@
 #define CG_BUILD_PARALLEL  @BUILDPARALLEL@
 
 #define CG_MAX_INT32 0x7FFFFFFF
 #ifdef _WIN32
 # define CG_LONG_T __int64
-#ifdef _MSC_VER
-# ifdef CG_BUILD_64BIT
-#  define stat _stat32i64
-# endif
-#endif
 #else
 # define CG_LONG_T @CGLONGT@
 #endif
 
 #if CG_BUILD_HDF5
 
