--- generate_vcxproj.bat	2018-03-19 05:26:56.000000000 -0800
+++ generate_vcxproj.bat	2026-02-10 11:57:36.000000000 -0800
@@ -49,13 +49,16 @@
 
 set _vcver_=%1
 
 set _clver_=1600
 set _vstoolset_=v100
 
-if "%_vcver_%"=="15.0" (
+if "%_vcver_%"=="17.0" (
+	set _clver_=1944
+	set _vstoolset_=v143
+) else if "%_vcver_%"=="15.0" (
 	set _clver_=1910
 	set _vstoolset_=v141
 ) else if "%_vcver_%"=="14.0" (
 	set _clver_=1900
 	set _vstoolset_=v140
 ) else ( if "%_vcver_%"=="12.0" (
@@ -108,12 +111,13 @@
 echo Examples:
 echo    generate_vcxproj 10.1 32 gdal_vs2010
 echo    generate_vcxproj 11.0 64 gdal_vs2012
 echo    generate_vcxproj 12.0 64 gdal_vs2013
 echo    generate_vcxproj 14.0 64 gdal_vs2015
 echo    generate_vcxproj 15.0 64 gdal_vs2017
+echo    generate_vcxproj 17.0 64 gdal_vs2022
 
 goto :end
 
 ::  *********************
 ::  Set Project file names
 ::  *********************
