--- src\H5Part.h	2026-02-09 10:11:42.000000000 -0800
+++ src\H5Part.h	2017-02-01 12:27:19.000000000 -0800
@@ -1,9 +1,12 @@
 #ifndef _H5Part_H_
 #define _H5Part_H_
 
+
+#include "h5part_exports.h"
+
 #include <stdlib.h>
 #include <stdarg.h>
 #include <hdf5.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -58,222 +61,250 @@
 #define H5PART_FLOAT64		((h5part_int64_t)H5T_NATIVE_DOUBLE)
 #define H5PART_FLOAT32		((h5part_int64_t)H5T_NATIVE_FLOAT)
 #define H5PART_CHAR		((h5part_int64_t)H5T_NATIVE_CHAR)
 #define H5PART_STRING		((h5part_int64_t)H5T_C_S1)
 
 /*========== File Opening/Closing ===============*/
-H5PartFile*
+H5PartFile H5PART_API *
 H5PartOpenFile(
 	const char *filename,
 	const char flags
 	);
 
-H5PartFile*
+H5PartFile H5PART_API *
 H5PartOpenFileAlign(
 	const char *filename,
 	const char flags,
 	h5part_int64_t align
 	);
 
 #define H5PartOpenFileSerial(x,y) H5PartOpenFile(x,y)
 #define H5PartOpenFileSerialAlign(x,y,z) H5PartOpenFileAlign(x,y,z)
 
 #ifdef PARALLEL_IO
-H5PartFile*
+H5PartFile H5PART_API *
 H5PartOpenFileParallel (
 	const char *filename,
 	const char flags,
 	MPI_Comm communicator
 	);
 
-H5PartFile*
+H5PartFile H5PART_API *
 H5PartOpenFileParallelAlign (
 	const char *filename,
 	const char flags,
 	MPI_Comm communicator,
 	h5part_int64_t align
 	);
 #endif
 
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartCloseFile (
 	H5PartFile *f
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartFileIsValid (
 	H5PartFile *f
 	);
 
 /*============== File Writing Functions ==================== */
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartDefineStepName (
 	H5PartFile *f,
 	const char *name,
 	const h5part_int64_t width
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartSetNumParticles ( 
 	H5PartFile *f, 
 	const h5part_int64_t nparticles
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartSetNumParticlesStrided (
 	H5PartFile *f,				/*!< [in] Handle to open file */
 	const h5part_int64_t nparticles,	/*!< [in] Number of particles */
 	const h5part_int64_t stride		/*!< [in] Stride (e.g. number of fields in the particle array) */
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartSetChunkSize (
 	H5PartFile *f,
 	const h5part_int64_t size
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartWriteDataFloat64 (
 	H5PartFile *f,
 	const char *name,
 	const h5part_float64_t *array
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartWriteDataFloat32 (
 	H5PartFile *f,
 	const char *name,
 	const h5part_float32_t *array
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartWriteDataInt64 (
 	H5PartFile *f,
 	const char *name,
 	const h5part_int64_t *array
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartWriteDataInt32 (
 	H5PartFile *f,
 	const char *name,
 	const h5part_int32_t *array
 	);
 
+h5part_int64_t H5PART_API
+H5PartAppendDataFloat64 (
+	H5PartFile *f,
+	const char *name,
+	const h5part_float64_t *array
+	);
+
+h5part_int64_t H5PART_API
+H5PartAppendDataFloat32 (
+	H5PartFile *f,
+	const char *name,
+	const h5part_float32_t *array
+	);
+
+h5part_int64_t H5PART_API
+H5PartAppendDataInt64 (
+	H5PartFile *f,
+	const char *name,
+	const h5part_int64_t *array
+	);
+
+h5part_int64_t H5PART_API
+H5PartAppendDataInt32 (
+	H5PartFile *f,
+	const char *name,
+	const h5part_int32_t *array
+	);
+
 /*================== File Reading Routines =================*/
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartSetStep (
 	H5PartFile *f,
 	const h5part_int64_t step
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartHasStep (
 	H5PartFile *f,
 	const h5part_int64_t step
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartGetNumSteps (
 	H5PartFile *f
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartGetNumDatasets (
 	H5PartFile *f
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartGetDatasetName (
 	H5PartFile *f,
 	const h5part_int64_t idx,
 	char *name,
 	const h5part_int64_t maxlen
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartGetDatasetInfo (
 	H5PartFile *f,
 	const h5part_int64_t idx,
 	char *name,
 	const h5part_int64_t maxlen,
 	h5part_int64_t *type,
 	h5part_int64_t *nelem);
 
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartGetNumParticles (
 	H5PartFile *f
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartSetView (
 	H5PartFile *f,
 	const h5part_int64_t start,
 	const h5part_int64_t end
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartSetViewIndices (
 	H5PartFile *f,			/*!< [in]  Handle to open file */
 	const h5part_int64_t *indices,	/*!< [in]  List of indices */
 	h5part_int64_t nelems		/*!< [in]  Size of list */
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartGetView (
 	H5PartFile *f,
 	h5part_int64_t *start,
 	h5part_int64_t *end
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartHasView (
 	H5PartFile *f
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartResetView (
 	H5PartFile *f
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartSetCanonicalView (
 	H5PartFile *f
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartReadDataFloat64(
 	H5PartFile *f,
 	const char *name,
 	h5part_float64_t *array
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartReadDataFloat32(
 	H5PartFile *f,
 	const char *name,
 	h5part_float32_t *array
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartReadDataInt64 (
 	H5PartFile *f,
 	const char *name,
 	h5part_int64_t *array
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartReadDataInt32 (
 	H5PartFile *f,
 	const char *name,
 	h5part_int32_t *array
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartReadParticleStep (
 	H5PartFile *f,
 	const h5part_int64_t step,
 	h5part_float64_t *x, /* particle positions */
 	h5part_float64_t *y,
 	h5part_float64_t *z,
@@ -288,131 +319,131 @@
    and step attributes which are bound to the current timestep.  You 
    must set the timestep explicitly before writing the attributes (just
    as you must do when you write a new dataset.  Currently there are no
    attributes that are bound to a particular data array, but this could
    easily be done if required.
 */
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartWriteStepAttrib (
 	H5PartFile *f,
 	const char *attrib_name,
 	const h5part_int64_t attrib_type,
 	const void *attrib_value,
 	const h5part_int64_t attrib_nelem
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartWriteFileAttrib (
 	H5PartFile *f,
 	const char *attrib_name,
 	const h5part_int64_t attrib_type,
 	const void *attrib_value,
 	const h5part_int64_t attrib_nelem
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartWriteFileAttribString (
 	H5PartFile *f,
 	const char *name,
 	const char *value
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartWriteStepAttribString ( 
 	H5PartFile *f,
 	const char *name,
 	const char *value
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartGetNumStepAttribs ( /* for current filestep */
 	H5PartFile *f
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartGetNumFileAttribs (
 	H5PartFile *f
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartGetStepAttribInfo (
 	H5PartFile *f,
 	const h5part_int64_t attrib_idx,
 	char *attrib_name,
 	const h5part_int64_t len_of_attrib_name,
 	h5part_int64_t *attrib_type,
 	h5part_int64_t *attrib_nelem
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartGetFileAttribInfo (
 	H5PartFile *f,
 	const h5part_int64_t idx,
 	char *name,
 	const h5part_int64_t maxnamelen,
 	h5part_int64_t *type,
 	h5part_int64_t *nelem
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartReadStepAttrib (
 	H5PartFile *f,
 	const char *name,
 	void *value
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartReadFileAttrib (
 	H5PartFile *f,
 	const char *name,
 	void *value
 	);
 
 /*============ Error Reporting and Configuration =============*/
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartSetVerbosityLevel (
 	const h5part_int64_t level
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartSetThrottle (
 	H5PartFile *f,
 	int factor
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartSetErrorHandler (
 	const h5part_error_handler handler
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartGetErrno (
 	void
 	);
 
-h5part_error_handler
+h5part_error_handler H5PART_API
 H5PartGetErrorHandler (
 	void
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartReportErrorHandler (
 	const char *funcname,
 	const h5part_int64_t eno,
 	const char *fmt,
 	...
 	);
 
-h5part_int64_t
+h5part_int64_t H5PART_API
 H5PartAbortErrorHandler (
 	const char *funcname,
 	const h5part_int64_t eno,
 	const char *fmt,
 	...
 	);
 
 #ifdef __cplusplus
 }
 #endif
 
 #endif
