--- src\H5PartPrivate.h	2011-09-19 13:19:24.000000000 -0800
+++ src\H5PartPrivate.h	2026-02-24 19:19:58.000000000 -0800
@@ -3,13 +3,13 @@
 
 #if H5_VERS_MAJOR == 1 && H5_VERS_MINOR == 6
 #define H5PART_USE_HDF5_16
 #define H5_USE_16_API
 #endif
 
-#if H5_VERS_MAJOR == 1 && H5_VERS_MINOR >= 8
+#if H5_VERS_MAJOR == 2 || (H5_VERS_MAJOR == 1 && H5_VERS_MINOR >= 8)
 #define H5PART_HAVE_HDF5_18
 #endif
 
 #define H5PART_GROUPNAME_STEP	"Step"
 
 #define H5PART_SET_STEP_READ_ONLY 0
